sound/soc/sof/topology.c

```diff
@@ -1233,6 +1240,60 @@ static int sof_control_unload(struct snd_soc_component *scomp,
  * DAI Topology
  */
 
+static int sof_core_enable(struct snd_sof_dev *sdev, int core)
+{
+	struct sof_ipc_pm_core_config pm_core_config = {
+		.hdr = {
+			.cmd = SOF_IPC_GLB_PM_MSG | SOF_IPC_PM_CORE_ENABLE,
+			.size = sizeof(pm_core_config),
+		},
+		.enable_mask = sdev->enabled_cores_mask | 1 << core,
+	};
+	int ret;
+
+	if (sdev->enabled_cores_mask & 1 << core)
+		return 0;
+
+	/* power up the core */
+	ret = snd_sof_dsp_core_power_up(sdev, 1 << core);
+	if (ret < 0) {
+		dev_err(sdev->dev, "error: %d powering up core %d\n",
+			ret, core);
+		return ret;
+	}
+
+	/* update enabled cores mask */
+	sdev->enabled_cores_mask |= 1 << core;
+
+	/* Now notify DSP that the core has been powered up */
+	ret = sof_ipc_tx_message(sdev->ipc, pm_core_config.hdr.cmd,
+				 &pm_core_config, sizeof(pm_core_config),
+				 &pm_core_config, sizeof(pm_core_config));
+	if (ret < 0)
+		dev_err(sdev->dev, "error: core %d enable ipc failure %d\n",
+			core, ret);
+
+	return ret;
```

Comment:
@lyakh can we just return sof_ipc_tx_message() here? There will be an error printed in that function anyway no?

Reply:
@ranj063 well not in all cases. Some failures don't print an error message. Those are really exotic ones like invalid pointers etc. Some calls to it print an error message, some don't. I'd keep it for now, but can remove it too if it's really undesired.